### PR TITLE
Serve mock index.html and window.context

### PR DESCRIPTION
**Pull request by felixfbecker**
We currently need to record a HAR for every path, despite those serving the same index.html with only minor differences like the `<title>`. This could be annoying to record for pages requiring backend state, like the index.html for a campaigns details page. We also need the ability to stub  `window.context`, in a type-safe way.
This makes us always serve a mock index.html, which removes the need for recording HARs on the webapp. HARs will still be used for the browser extension though.

Also added `NODE_NO_WARNINGS` for now because Polly causes a lot of `uncaughtException` warnings and `web-ext`'s transitive dependency of an old version of `shelljs` causes circular dependency warnings.  I don't like silencing warnings, but they were _extremely_ verbose (make the test output barely readable), and are outside of our control. We have an ESLint rule to catch uncaught exceptions.

### CODE REVIEW

File: web/package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "license": "MIT",
   "scripts": {
     "test": "jest --testPathIgnorePatterns end-to-end --testPathIgnorePatterns regression integration",
-    "test:integration": "mocha ./src/integration/**/*.test.ts",
+    "test:integration": "NODE_NO_WARNINGS=1 mocha ./src/integration/**/*.test.ts",
```
Comment (eseliger) [inline web/package.json:9]: why is that required?

Comment (felixfbecker) [reply web/package.json:9]: Added to the PR description

Comment (eseliger) [reply web/package.json:9]: Thanks!
